frontend/src/metabase/query_builder/actions/native.js

```diff
@@ -154,41 +147,15 @@ export const insertSnippet = snip => (dispatch, getState) => {
 export const SET_TEMPLATE_TAG = "metabase/qb/SET_TEMPLATE_TAG";
 export const setTemplateTag = createThunkAction(
   SET_TEMPLATE_TAG,
-  templateTag => {
+  (tag, parameter) => {
     return (dispatch, getState) => {
-      const {
-        qb: { card, uiControls },
-      } = getState();
-
-      const updatedCard = Utils.copy(card);
-
-      // when the query changes on saved card we change this into a new query w/ a known starting point
-      if (uiControls.queryBuilderMode !== "dataset" && updatedCard.id) {
-        delete updatedCard.id;
-        delete updatedCard.name;
-        delete updatedCard.description;
-      }
-
-      // we need to preserve the order of the keys to avoid UI jumps
-      const updatedTagsCard = updateIn(
-        updatedCard,
-        ["dataset_query", "native", "template-tags"],
-        tags => {
-          const { name } = templateTag;
-          const newTag =
-            tags[name] && tags[name].type !== templateTag.type
-              ? // when we switch type, null out any default
-                { ...templateTag, default: null }
-              : templateTag;
-          return { ...tags, [name]: newTag };
-        },
-      );
-
-      return assoc(
-        updatedTagsCard,
-        "parameters",
-        getTemplateTagParameters(getTemplateTagsForParameters(updatedTagsCard)),
-      );
+      const question = getQuestion(getState());
+      const newQuestion = question
+        .query()
+        .setTemplateTag(tag.name, tag, parameter)
+        .question();
+
+      dispatch(updateQuestion(newQuestion));
```

Comment:
`updateQuestion` handles all this complicated logic in the removed block

Reply:
Unless I misunderstand this, I think we can remove the `createThunkAction` call and just return the `(dispatch, getState) => xxx` function directly, since previously we will use the return result from this function as a payload for `SET_TEMPLATE_TAG`, but it isn't used anymore (the reducer block that used this action is deleted in this PR).